opal/corelib/struct.rb

```diff
@@ -83,7 +83,15 @@ def [](name)
 
       name = members[name]
     elsif String === name
-      raise NameError.new("no member '#{name}' in struct", name) unless members.include?(name.to_sym)
+      %x{
+        if(!self.$$data.hasOwnProperty(name)) {
+          Opal.Kernel.$raise(
+            Opal.NameError.$new(
+              "no member '" + name + "' in struct", name
+            )
```

Comment:
s/perfect/performance 

Yeah, "night time is the right time" :)
Just a stylistic note anyway, the code looks correct. 👍🏼